### DISCUSSION TITLE
How does Solid's "semi-synthetic event delegation" work?

### DISCUSSION THREAD

**Discussion by jorroll** [solidjs/solid#1548] [Q&A]
In the Solidjs documentation for [the `on___` JSX attribute](https://www.solidjs.com/docs/latest/api#on___), it says

> Solid uses semi-synthetic event delegation for common UI events that are composed and bubble.

Is there any documentation for these semi-synthetic events (I couldn't find any on the website)? How do they work? How are these events different from native events? Which events are semi-synthetic and which aren't?  React's synthetic event listeners work very differently from native event listeners. Is this true in Solidjs as well?

Thanks!

**Comment by atk** [comment]
It sets the handlers on the mount root by using the delegateEvents function, captures events where necessary and then walks through all elements with a $$click (for click events, change name for other events) and defers the event to them.

It doesn't do as much as React does, like flipping onInput and onChange and adding different other properties.

**Reply by jorroll** [reply]
Thanks! Is this only documented in the source code? The Solidjs docs indicate that only "common UI events" use event delegation like this. Do you know which events use it and which do not? 

Do you know what sort of performance benefit these delegated events provide? In my experience with React, delegated events work fine until you start mixing them with non-delegated event listeners--at which point the delegated events become very annoying. I'm curious as to why/when event delegation becomes necessary in Solidjs.

**Reply by maxmilton** [reply]
Probably this set of event names: <https://github.com/ryansolid/dom-expressions/blob/e1bbcce54512f8eb50632abe9bb393dc01dfc759/packages/dom-expressions/src/constants.js#L63-L86>
